From 02078903482aeaef37566d589b8a8872e787d8db Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Fri, 14 Jan 2022 17:14:54 +0100
Subject: [PATCH 58/67] Revert "bfq: fix use-after-free in
 bfq_dispatch_request"

This reverts commit 6ea822e67a26e82fe03b9f9eb9976f4dbf07da61.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 block/bfq-iosched.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index 9c2e0b027..bde7cf528 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -5232,7 +5232,6 @@ static struct request *bfq_dispatch_request(struct blk_mq_hw_ctx *hctx)
 	spin_lock_irq(&bfqd->lock);
 
 	in_serv_queue = bfqd->in_service_queue;
-	in_serv_queue->ref++; /* aviod in_serv_queue release */
 	waiting_rq = in_serv_queue && bfq_bfqq_wait_request(in_serv_queue);
 
 	rq = __bfq_dispatch_request(hctx);
@@ -5244,10 +5243,6 @@ static struct request *bfq_dispatch_request(struct blk_mq_hw_ctx *hctx)
 
 	bfq_update_dispatch_stats(hctx->queue, rq, in_serv_queue,
 				  idle_timer_disabled);
-	/* resume in_serv_queue */
-	spin_lock_irq(&bfqd->lock);
-	bfq_put_queue(in_serv_queue);
-	spin_unlock_irq(&bfqd->lock);
 
 	return rq;
 }
-- 
2.34.1.75.gabe6bb3905

